src/Umbraco.Core/Services/Implement/ContentService.cs

```diff
@@ -1218,16 +1294,46 @@ public IEnumerable<PublishResult> SaveAndPublishBranch(IContent content, bool fo
             // note: EditedValue and PublishedValue are objects here, so it is important to .Equals()
             // and not to == them, else we would be comparing references, and that is a bad thing
 
-            bool IsEditing(IContent c, string l)
-                => c.PublishName != c.Name ||
-                   c.PublishedCultures.Where(x => x.InvariantEquals(l)).Any(x => c.GetCultureName(x) != c.GetPublishName(x)) ||
-                   c.Properties.Any(x => x.Values.Where(y => culture == "*" || y.Culture.InvariantEquals(l)).Any(y => !y.EditedValue.Equals(y.PublishedValue)));
+            // determines whether the document is edited, and thus needs to be published,
+            // for the specified culture (it may be edited for other cultures and that
+            // should not trigger a publish).
+            HashSet<string> ShouldPublish(IContent c)
+            {
+                var isRoot = c.Id == content.Id;
+
+                if (c.ContentType.VariesByCulture())
+                {
+                    // variant content type
+                    // add culture if edited, and already published or forced
+                    if (c.IsCultureEdited(culture) && (c.IsCulturePublished(culture) || force || isRoot))
```

Comment:
@zpqrtbnk I've pushed a test for this that fails but it should work. If I have no changes to the root document but I have pending changes to any descendants, those descendants should still be published.